src/crypto/public_key/elliptic/indutnyKey.js

```diff
@@ -0,0 +1,35 @@
+// OpenPGP.js - An OpenPGP implementation in javascript
+// Copyright (C) 2015-2016 Decentral
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 3.0 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+/**
+ * @fileoverview Wrapper for a KeyPair of an curve from indutny/elliptic library
+ * @requires enums
+ * @requires asn1.js
+ * @module crypto/public_key/elliptic/indutnyKey
+ */
+
+/**
+ * @constructor
+ */
+function KeyPair(indutnyCurve, options) {
+  this.keyPair = indutnyCurve.keyPair(options);
+  if (this.keyPair.validate().result !== true) {
+    throw new Error('Invalid elliptic public key');
+  }
```

Comment:
I'm not sure that we should always call `validate` here - before, we were only calling it in `keyFromPublic`, now we're also calling it in what used to be `keyFromPrivate`. Could you maybe briefly test how long `validate` takes to see whether it makes any difference at all?

Reply:
I wanted to explicitly cache `indutnyCurve` here. I know that nodeJs is caching require, however I still think it is useful to have explicit cache, for various browser environments.